backend/lib/deal-observer.js

```diff
@@ -31,6 +31,16 @@ export async function fetchDealWithHighestActivatedEpoch (pgPool) {
   return result.length > 0 ? result[0] : null
 }
 
+/**
+ * @param {Queryable} pgPool
+ * @returns {Promise<number>}
+ * */
+export async function fetchNumberOfStoredActiveDeals (pgPool) {
+  const query = 'SELECT COUNT(*) FROM active_deals'
+  const result = await pgPool.query(query)
+  return result.rows.length > 0 ? result.rows[0].count : 0
```

Comment:
Can this query ever have no rows? I believe it's safe to return this
```suggestion
  return result.rows[0].count
```

Reply:
I was not sure, from an SQL perspective it should not happen but I was not sure about the internals of pgPool.